### PR TITLE
Fix symfony bc break

### PR DESCRIPTION
**Pull request by Deepress**
Properly formatted fix solution

### CODE REVIEW

**Comment by Majkl578** [comment]
Note: Deprecations are not BC breaks.

File: src/DI/ValidatorExtension.php [hunk 1]
```diff
@@ -37,7 +37,8 @@ class ValidatorExtension extends Nette\DI\CompilerExtension implements ITranslat
 		'cache' => 'default',
 		'translationDomain' => NULL,
 		'debug' => '%debugMode%',
-		'strictEmail' => FALSE,
+		'strictEmail' => NULL,
+        'emailValidationMode' => \Symfony\Component\Validator\Constraints\Email::VALIDATION_MODE_LOOSE
```
Comment (enumag) [inline src/DI/ValidatorExtension.php]: Wrong indentation, missing comma, should use the import.

File: src/DI/ValidatorExtension.php [hunk 2]
```diff
@@ -80,12 +81,18 @@ public function loadConfiguration()
 			->setClass('Symfony\Component\Validator\Validator\ValidatorInterface')
 			->setFactory('Symfony\Component\Validator\Validator\RecursiveValidator');
 
-		Validators::assertField($config, 'strictEmail', 'boolean');
+		if ($config['strictEmail'] !== NULL) {
+            Validators::assertField($config, 'strictEmail', 'boolean');
+            trigger_error('`strictEmail` configuration option is deprecated, use `emailValidationMode` instead.', E_USER_DEPRECATED);
+            $config['emailValidationMode'] = Email::VALIDATION_MODE_LOOSE;
+        }
```
Comment (enumag) [inline src/DI/ValidatorExtension.php]: Wrong indentation.

File: src/DI/ValidatorExtension.php [hunk 3]
```diff
@@ -80,12 +81,18 @@ public function loadConfiguration()
...
+
+        Validators::assertField($config, 'emailValidationMode', 'string');
```
Comment (enumag) [inline src/DI/ValidatorExtension.php]: Wrong indentation.

File: tests/KdybyTests/Validator/Extension.phpt [hunk 4]
```diff
@@ -118,9 +118,9 @@ class ExtensionTest extends Tester\TestCase
 	public function strictEmailDataProvider()
 	{
 		return [
-			[[], FALSE],
-			[[__DIR__ . '/config/strict-email.neon'], TRUE],
-			[[__DIR__ . '/config/non-strict-email.neon'], FALSE],
+			[[], 'loose'],
+			[[__DIR__ . '/config/strict-email.neon'], 'strict'],
+			[[__DIR__ . '/config/non-strict-email.neon'], 'loose'],
```
Comment (enumag) [inline tests/KdybyTests/Validator/Extension.phpt]: Use constants instead.

File: src/DI/ValidatorExtension.php [hunk 5]
```diff
@@ -80,12 +81,18 @@ public function loadConfiguration()
 			->setClass('Symfony\Component\Validator\Validator\ValidatorInterface')
 			->setFactory('Symfony\Component\Validator\Validator\RecursiveValidator');
 
-		Validators::assertField($config, 'strictEmail', 'boolean');
+		if ($config['strictEmail'] !== NULL) {
+            Validators::assertField($config, 'strictEmail', 'boolean');
+            trigger_error('`strictEmail` configuration option is deprecated, use `emailValidationMode` instead.', E_USER_DEPRECATED);
+            $config['emailValidationMode'] = Email::VALIDATION_MODE_LOOSE;
```
Comment (enumag) [inline src/DI/ValidatorExtension.php]: This line can be removed, `Email::VALIDATION_MODE_LOOSE` is default anyway.

**Comment by Deepress** [comment]
Applied request changes in latest commit

**Comment by enumag** [comment]
Looks good at first glance. Travis failures are most likely related to outdated dependencies. I believe we can drop PHP <7.0 and Nette <2.4 now. This will need some changes in composer.json and travis.yml.

**Pull request merged**